doc/design/bucketpath.rst

```diff
@@ -0,0 +1,485 @@
+
+- Embrace and use ``pathlib`` semantics and naming conventions when applicable.
+- For significant semantic deviations, opt for unique, clear names that avoid confusion with ``pathlib`` terminology.
+
+
+.. Attention::
+
+   The subsequent subsections include code snippets intended primarily for the implementer's reference.
+   Therefore, comments and docstrings within the sudo code may need adaptation for the actual implementation.
+
+
+Pathlike
+--------
+
+.. code-block:: python
+
+    from typing import Protocol
+
+    class Pathlike(Protocol):
+
+        @property
+        def name:
+            """
+            A string representing the final path component, excluding the drive and root, if any.
+            """
+
+        @property
+        def suffix:
+            """
+            The file extension of the final component, if any.
+            """
+
+        @property
+        def root:
+            """
+            A string representing the root, if any.
+            """
+
+        @property
+        def parent:
+            """
+            The logical parent of this path.
+            """
+
+        def as_uri():
+            """
+            Represent the path as a file URI. Can be used to reconstruct the location/path.
+            """
+
+        def exists():
+            """
+            Return True if the path points to an existing file or directory.
+            """
+
+        def is_dir():
+            """
+            Return True if the path points to a directory, False if it points to another kind of file.
+            """
+
+        def is_file():
+            """
+            Return True if the path points to a regular file, False if it points to another kind of file.
+            """
+
+        def read(chunk_size: int = 8192) -> Iterable[ByteString]:
+            """
+            Read the content of a the file behind this path.
+
+            Only works for pathslike objects which return True for `is_file()`.
+
+            Args:
+                chunk_size: which will be yieled by the iterator.
+
+            Returns:
+                Returns an iterator which can be used to read the contents of the path in chunks.
+
+            Raises:
+                NotAFileError: if the pathlike object does not point to a file.
+                FileNotFoundError: If the file does not exist.
+            """
+
+        def write(data: ByteString | BinaryIO | Iterable[ByteString]):
+            """
+            Writes data to a this path. 
+
+            After successfully writing to this path `exists` will yield true for this path.
+
+            Args:
+                data: which shall be writen to the path.
+
+
+            Raises:
+                FileExistsError: if the file already exists.
+                NotAFileError: if the pathlike object is not a file path.
+
+                Note: 
+                If you still need to write the new content, please delete the current contents first,
```

Comment:
this I don't get, overwriting is possible

Reply:
This is a challenging design decision. Implementing overwriting functionality is feasible; however, it necessitates the deletion of the existing file in the Bucket. This process complicates the provision of a stable API, as file deletion requests do not immediately remove the file from BucketFS.

Reply:
overwrite should work without problems